src/video-custom-controls.js

```diff
@@ -0,0 +1,603 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Animation} from './animation';
+import {dev} from './log';
+import {listen} from './event-helper';
+import {Services} from './services';
+import {VideoEvents} from './video-interface';
+import {formatTime} from './utils/datetime';
+import * as st from './style';
+import * as tr from './transition';
+
+export class CustomControls {
+
+  constructor(
+    ampdoc,
+    entry,
+    darkSkin = false,
+    mainCtrls = ['time', 'spacer', 'volume', 'fullscreen'],
+    miniCtrls = ['play', 'volume', 'fullscreen'],
+    floating = 'play'
+  ) {
+
+    /** @const {!./service/ampdoc-impl.AmpDoc}  */
+    this.ampdoc_ = ampdoc;
+
+    /** @private {!./service/video-manager-impl.VideoEntry} */
+    this.entry_ = entry;
+
+    /** @private @const {!./service/vsync-impl.Vsync} */
+    this.vsync_ = Services.vsyncFor(this.ampdoc_.win);
+
+    /** @private {boolean} */
+    this.darkSkin_ = darkSkin;
+
+    /** @private {Element} */
```

Comment:
Either `!Element` or `?Element`. Only primitives are implicitly `!`, so they don't need it.